### DISCUSSION TITLE
Is remark-toc suppose to ignore markdown custom ids for headings?

### DISCUSSION THREAD

**Discussion by Expoverse** [remarkjs/remark#1124] [Q&A] [answered]
Using remark-toc doesn't support custom ids. I'm not sure if this is supposed to happen, since if you are rendering markdown with a table of contents using remark-toc, then you should be allowed to set the heading ids in markdown. 

```
Setting Up a Demonstration {#setting-up-a-demonstration}
```

Generates this?

```
-   [Setting Up a Demonstration {#setting-up-a-demonstration}](#setting-up-a-demonstration-setting-up-a-demonstration)
-   [Methods](#methods)
    -   [Using the Float Method](#using-the-float-method)
    -   [Using Grid Method](#using-grid-method)
    -   [Using Flexbox Method](#using-flexbox-method)
    -   [Using display: inline Method](#using-display-inline-method)
-   [Summary](#summary)
-   [Recommended Articles](#recommended-articles)
```

**Comment by ChristianMurphy** [comment, marked as answer]
It is not supported by `remark-toc`.
There is no custom id in commonmark with the syntax you show, nor is it in gfm or mdx.

It could be supported by adding https://github.com/imcuttle/remark-heading-id or https://github.com/iloveip/remark-heading-autoid
Then building your own TOC-like plugin for the different and custom system you are interested in (you can learn more about creating plugins here https://unifiedjs.com/learn/)
